code/src/context/ModalContext.tsx

```diff
@@ -0,0 +1,57 @@
+import {
+  ReactElement,
+  ReactNode,
+  createContext,
+  useContext,
+  useRef,
+  useState
+} from "react";
+import Dialog from "@mui/material/Dialog";
+import AppBox from "@/components/app-box/AppBox";
+
+type ModalContextType = {
+  handleOpenModal: (component: ReactElement) => void;
+  handleCloseModal: () => void;
+};
+
+type ModalProviderProps = { children: ReactNode };
```

Comment:
It's look like `I18nProividerProps` type
Move this `type` to `common` types and reuse it. define only in one place

Reply:
used utility type from `React`
![image](https://github.com/idx-academy/spa-orders/assets/118522791/33ed4634-3533-45f5-a580-8c267216b26c)